src/main/scala/tech/cryptonomic/conseil/tezos/NodeSenderOperations.scala

```diff
@@ -0,0 +1,333 @@
+        )
+      case None =>
+        Map(
+          "branch" -> blockHead.data.header.predecessor,
+          "operations" -> operations
+        )
+    }
+
+    RpcHandler
+      .runPost("/blocks/head/proto/helpers/forge/operations", Some(toJson(payload)))
+      .map(json => fromJson[ForgedOperation](json).operation)
+  }
+
+  /**
+    * Signs a forged operation
+    * @param forgedOperation  Forged operation group returned by the Tezos client (as a hex string)
+    * @param keyStore         Key pair along with public key hash
+    * @return                 Bytes of the signed operation along with the actual signature
+    */
+  def signOperationGroup[Eff[_]: MonadThrow](forgedOperation: String, keyStore: KeyStore): Eff[SignedOperationGroup] =
+    for {
+      privateKeyBytes <- CryptoUtil.base58CheckDecode(keyStore.privateKey, "edsk").liftTo[Eff]
+      watermark = "03" // In the future, we must support "0x02" for endorsements and "0x01" for block signing.
+      watermarkedForgedOperationBytes = SodiumUtils.hex2Binary(watermark + forgedOperation)
+      hashedWatermarkedOpBytes = SodiumLibrary.cryptoGenerichash(watermarkedForgedOperationBytes, 32)
+      opSignature: Array[Byte] = SodiumLibrary.cryptoSignDetached(hashedWatermarkedOpBytes, privateKeyBytes.toArray)
+      hexSignature <- CryptoUtil.base58CheckEncode(opSignature.toList, "edsig").liftTo[Eff]
+      signedOpBytes = SodiumUtils.hex2Binary(forgedOperation) ++ opSignature
+    } yield SignedOperationGroup(signedOpBytes, hexSignature)
+
+  /**
+    * Computes the ID of an operation group using Base58Check.
+    * @param signedOpGroup  Signed operation group
+    * @return               Base58Check hash of signed operation
+    */
+  def computeOperationHash[Eff[_]: MonadThrow](signedOpGroup: SignedOperationGroup): Eff[String] =
+    Try(SodiumLibrary.cryptoGenerichash(signedOpGroup.bytes, 32)).flatMap { hash =>
+      CryptoUtil.base58CheckEncode(hash.toList, "op")
+    }.liftTo[Eff]
+
+  /**
+    * Applies an operation using the Tezos RPC client.
+    * @param blockHead            Block head
+    * @param operationGroupHash   Hash of the operation group being applied (in Base58Check format)
+    * @param forgedOperationGroup Forged operation group returned by the Tezos client (as a hex string)
+    * @param signedOpGroup        Signed operation group
+    * @return                     Array of contract handles
+    */
+  def applyOperation[Eff[_]: PostHandler: Functor](
+    blockHead: Block,
+    operationGroupHash: String,
+    forgedOperationGroup: String,
+    signedOpGroup: SignedOperationGroup
+  ): Eff[AppliedOperation] = {
+    val payload: AnyMap = Map(
+      "pred_block" -> blockHead.data.header.predecessor,
+      "operation_hash" -> operationGroupHash,
+      "forged_operation" -> forgedOperationGroup,
+      "signature" -> signedOpGroup.signature
+    )
+
+    RpcHandler.runPost("/blocks/head/proto/helpers/apply_operation", Some(toJson(payload))).map { result =>
```

Comment:
Where is the implicit RpcHandler handler instance here coming from?

Reply:
It comes from the method's type constraint
```
Eff[_]: PostHandler: Functor](...)
            ^^^
// this translates syntactically to an extra (implicit someName: PostHandler[Eff]) parameter
```

The `applyOperation` essentially delegates his dependency on the RpcHandler to the caller, and uses a simplified type alias (i.e. PostHandler) to make the method definition compact